web/src/features/datasets/server/dataset-router.ts

```diff
@@ -582,13 +602,12 @@ export const datasetRouter = createTRPCRouter({
         },
       });
 
-      const observationIds = runItems
-        .map((ri) => ri.observation?.id)
-        .filter(Boolean) as string[];
       const observations = await ctx.prisma.observationView.findMany({
         where: {
           id: {
-            in: observationIds,
+            in: runItems
+              .map((ri) => ri.observationId)
+              .filter(Boolean) as string[],
```

Comment:
i think adding a projectid filter here would not be harmful

Reply:
good one, added